skrub/_selectors/tests/test_selectors.py

```diff
@@ -0,0 +1,113 @@
+import inspect
+import pickle
+import types
+
+import pytest
+
+from skrub import _dataframe as sbd
+from skrub import _selectors as s
+
+
+def test_repr():
+    """
+    >>> from skrub import _selectors as s
+    >>> s.numeric() - s.boolean()
+    (numeric() - boolean())
+    >>> s.numeric() | s.glob("*_mm") - s.regex(r"^[ 0-9]+_mm$")
+    (numeric() | (glob('*_mm') - regex('^[ 0-9]+_mm$')))
+    >>> s.cardinality_below(30)
+    cardinality_below(30)
+    >>> s.string() | s.any_date() | s.categorical()
+    ((string() | any_date()) | categorical())
+    >>> s.float() & s.integer()
+    (float() & integer())
+
+    """
+
+
+def test_glob(df_module):
+    df = df_module.example_dataframe
+    assert s.glob("*").expand(df) == s.all().expand(df)
+    assert s.glob("xxx").expand(df) == []
+    assert (s.glob("[Ii]nt-*") | s.glob("?loat-col")).expand(df) == [
+        "int-col",
+        "float-col",
+    ]
+
+
+def test_regex(df_module):
+    df = df_module.example_dataframe
+    assert (s.regex("int-.*") | s.regex("float-") | s.regex("date-$")).expand(df) == [
+        "int-col",
+        "float-col",
+    ]
+
+
+def test_dtype_selectors(df_module):
+    df = df_module.example_dataframe
+    cat_col = sbd.rename(sbd.to_categorical(sbd.col(df, "str-col")), "cat-col")
+    df = sbd.make_dataframe_like(df, sbd.to_column_list(df) + [cat_col])
+    assert s.numeric().expand(df) == ["int-col", "float-col"]
+    assert (s.numeric() | s.boolean()).expand(df) == [
+        "int-col",
+        "float-col",
+        "bool-col",
+    ]
+    assert s.integer().expand(df) == ["int-col"]
+    assert s.float().expand(df) == ["float-col"]
+    assert s.string().expand(df) == ["str-col"]
+    assert s.categorical().expand(df) == ["cat-col"]
+
```

Comment:
```suggestion
assert s.any_date().expand(df) == ["datetime-col", "date-col"]

```

Reply:
pandas doesn't have a notion of "date", only datetime AFAICT

Reply:
I updated the test to include `any_date`, distinguishing the pandas & polars cases